dispaset/__init__.py

```diff
@@ -2,11 +2,19 @@
 import os
 
 # Sets the __version__ variable
-from ._version import __version__
 from .common import commons
 
-from .preprocessing.build import get_git_revision_tag
-__gitversion__ = get_git_revision_tag()
+try:
+    from setuptools_scm import get_version
+    version = get_version(version_scheme='post-release',
+                          local_scheme='dirty-tag',
+                          root='..', relative_to=__file__)
+except (ImportError, LookupError):
+    try:
+        from ._version import version
```

Comment:
I suggest keeping only one `_version `file (no release) as it can get confusing. this file will hold a default value, and should exist in the repository. It will get overwritten by `setuptools ` write_to

Reply:
I think there is a drawback to have only one auto-generated file dispaset/_version.py. You will never commit the good version to git.

Before commit, you have a dirty version like 2.3.post171+dirty and after, you have a hash version like 2.3.post172+g57f6e58 except you can't guess it in advance. Otherwise, I don't understand the use case.

Reply:
oof this is getting complicated...
I was thinking: We have a default _version.py in the repository containing the right major version e.g. 2.3

- If someone downloads it as zip then this is the version used in logs. We do not care about git commits in this case as the folder will not be a git repo
  - If he uses pip then the version used will be the fallback, which is _version.py
- If someone uses git to clone and pip install then the _version.py will be replaced with the right one, as write_to replaces the file. But I understand the confusion when you push that file that does not correspond to the real version. 

I leave it up to you, but if you think that 2 files are better give a more indicative name instead of release, e.g. fallback_version !